src/fsharp/NameResolution.fs

```diff
@@ -3260,38 +3260,45 @@ let rec ResolvePartialLongIdentInModuleOrNamespaceForRecordFields (ncenv: NameRe
     // get record type constructors
     let tycons = 
         mty.TypeDefinitions
-        |> List.filter (fun tcref -> not (tcref.LogicalName.Contains(",")))
-        |> List.filter (fun tycon -> tycon.IsRecordTycon)
-        |> List.filter (fun tycon -> not (IsTyconUnseen ad g ncenv.amap m (modref.NestedTyconRef tycon)))
+        |> List.filter (fun tcref -> 
+               (not (tcref.LogicalName.Contains(","))) &&
+               tcref.IsRecordTycon &&
+               (not (IsTyconUnseen ad g ncenv.amap m (modref.NestedTyconRef tcref))))
 
     let ilTyconNames = 
         mty.TypesByAccessNames.Values
-        |> Seq.toList
-        |> List.choose (fun (tycon:Tycon) -> if tycon.IsILTycon then Some tycon.DisplayName else None)
+        |> Seq.choose (fun (tycon:Tycon) -> if tycon.IsILTycon then Some tycon.DisplayName else None)
```

Comment:
so for seq we would theoretically use filter + map but for lists we would use choose or collect?